src/Features/Core/Portable/ExtractMethod/ExtractMethodResult.cs

```diff
@@ -64,5 +75,25 @@ internal class ExtractMethodResult
         /// internal status of result. more fine grained reason why it is failed. 
         /// </summary>
         internal OperationStatusFlag Status { get; }
+
+        public async Task<(Document document, SyntaxToken invocationNameToken)> GetFormattedDocumentAsync(CancellationToken cancellationToken)
+        {
+            if (DocumentWithoutFinalFormatting is null)
+                throw new InvalidOperationException();
+
+            var annotation = new SyntaxAnnotation();
+
+            var root = await DocumentWithoutFinalFormatting.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            root = root.ReplaceToken(InvocationNameToken, InvocationNameToken.WithAdditionalAnnotations(annotation));
+
+            var annotatedDocument = DocumentWithoutFinalFormatting.WithSyntaxRoot(root);
+            var simplifiedDocument = await Simplifier.ReduceAsync(annotatedDocument, Simplifier.Annotation, optionSet: null, cancellationToken).ConfigureAwait(false);
+            var simplifiedRoot = await simplifiedDocument.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var documentOptions = await DocumentWithoutFinalFormatting.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var formattedDocument = simplifiedDocument.WithSyntaxRoot(Formatter.Format(simplifiedRoot, Formatter.Annotation, DocumentWithoutFinalFormatting.Project.Solution.Workspace, documentOptions, FormattingRules, cancellationToken));
+            var formattedRoot = await formattedDocument.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            return (formattedDocument, formattedRoot.GetAnnotatedTokens(annotation).Single());
```

Comment:
would CodeAction.CleanupDocumentAsync be more appropriate here?

Reply:
I worry about unintended side effects. We only simplify and format explicitly annotated nodes to avoid alterations to portions of the document outside the refactoring scope.

Reply:
wfm.